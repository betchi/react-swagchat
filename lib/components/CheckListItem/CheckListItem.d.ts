/// <reference types="react" />
import * as React from 'react';
import { IUser } from 'swagchat-sdk';
import { IOnClickProps } from '../../';
export interface ICheckListItemProps extends IOnClickProps {
    name: string;
    pictureUrl: string;
    width?: number;
    height?: number;
    isChecked: boolean;
}
export declare class CheckListItem extends React.Component<ICheckListItemProps, void> {
    onClick(user: IUser): void;
    render(): JSX.Element;
}
