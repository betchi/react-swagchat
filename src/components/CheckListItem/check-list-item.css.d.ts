export const colors: string;
export const colorPrimary: string;
export const colorGreyAlpha: string;
export const root: string;
export const avatar: string;
export const subject: string;
export const flex1: string;
export const flex2: string;
export const flex3: string;
export const checked: string;
export const unchecked: string;
