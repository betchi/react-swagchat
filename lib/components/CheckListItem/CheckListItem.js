"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var _1 = require("../../");
var CheckListItem = (function (_super) {
    __extends(CheckListItem, _super);
    function CheckListItem() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    CheckListItem.prototype.onClick = function (user) {
        if (this.props.onClick) {
            this.props.onClick(user);
        }
    };
    CheckListItem.prototype.render = function () {
        return (React.createElement("div", { className: "check-list-item-root", onClick: this.onClick.bind(this, this.props.user) },
            React.createElement("div", { className: "check-list-item-flex1" },
                React.createElement(_1.Avatar, { src: this.props.user.pictureUrl ? this.props.user.pictureUrl : '', className: "check-list-item-avatar" })),
            React.createElement("div", { className: "check-list-item-flex2" },
                React.createElement("div", { className: "check-list-item-subject" }, this.props.user.name)),
            React.createElement("div", { className: "check-list-item-flex3" },
                React.createElement(_1.Button, { icon: React.createElement(_1.CheckCircle, null) }))));
    };
    return CheckListItem;
}(React.Component));
exports.CheckListItem = CheckListItem;
//# sourceMappingURL=CheckListItem.js.map