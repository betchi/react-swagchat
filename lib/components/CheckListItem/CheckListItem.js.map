{"version":3,"file":"CheckListItem.js","sourceRoot":"","sources":["../../../src/components/CheckListItem/CheckListItem.tsx"],"names":[],"mappings":";;;;;;;;;;;;AAAA,6BAA+B;AAE/B,2BAAiG;AAUjG;IAAmC,iCAA0C;IAA7E;;IA4BA,CAAC;IA3BC,+BAAO,GAAP,UAAQ,IAAW;QACjB,EAAE,CAAC,CAAC,IAAI,CAAC,KAAK,CAAC,OAAO,CAAC,CAAC,CAAC;YACvB,IAAI,CAAC,KAAK,CAAC,OAAO,CAAC,IAAI,CAAC,CAAC;QAC3B,CAAC;IACH,CAAC;IAED,8BAAM,GAAN;QAAA,iBAoBC;QAnBC,MAAM,CAAC,CACL,6BAAK,SAAS,EAAC,sBAAsB,EAAC,OAAO,EAAE,IAAI,CAAC,KAAK,CAAC,OAAO;YAC/D,6BAAK,SAAS,EAAC,uBAAuB,EAAC,KAAK,EAAE,EAAC,MAAM,EAAE,IAAI,CAAC,KAAK,CAAC,KAAK,GAAG,IAAI,EAAE,UAAU,EAAE,IAAI,CAAC,KAAK,CAAC,KAAK,GAAG,IAAI,EAAC;gBAClH,oBAAC,SAAM,IAAC,GAAG,EAAE,IAAI,CAAC,KAAK,CAAC,UAAU,EAAE,KAAK,EAAE,IAAI,CAAC,KAAK,CAAC,KAAK,EAAE,MAAM,EAAE,IAAI,CAAC,KAAK,CAAC,MAAM,GAAI,CACtF;YACN,6BAAK,SAAS,EAAC,uBAAuB,EAAC,KAAK,EAAE,EAAC,MAAM,EAAE,IAAI,CAAC,KAAK,CAAC,KAAK,GAAG,IAAI,EAAE,UAAU,EAAE,IAAI,CAAC,KAAK,CAAC,KAAK,GAAG,IAAI,EAAC;gBAClH,6BAAK,SAAS,EAAC,yBAAyB,IAAE,IAAI,CAAC,KAAK,CAAC,IAAI,CAAO,CAC5D;YACN,6BAAK,SAAS,EAAC,uBAAuB,EAAC,KAAK,EAAE,EAAC,MAAM,EAAE,IAAI,CAAC,KAAK,CAAC,KAAK,GAAG,IAAI,EAAE,UAAU,EAAE,IAAI,CAAC,KAAK,CAAC,KAAK,GAAG,IAAI,EAAC,IACjH,CAAC;gBACA,EAAE,CAAC,CAAC,KAAI,CAAC,KAAK,CAAC,SAAS,CAAC,CAAC,CAAC;oBACzB,MAAM,CAAC,oBAAC,SAAM,IAAC,IAAI,EAAE,oBAAC,qBAAkB,OAAG,GAAI,CAAC;gBAClD,CAAC;gBAAC,IAAI,CAAC,CAAC;oBACN,MAAM,CAAC,oBAAC,SAAM,IAAC,IAAI,EAAE,oBAAC,uBAAoB,IAAC,KAAK,EAAE,EAAC,IAAI,EAAE,0BAA0B,EAAC,GAAI,GAAI,CAAC;gBAC/F,CAAC;YACH,CAAC,CAAC,EAAE,CACA,CACF,CACP,CAAC;IACJ,CAAC;IACH,oBAAC;AAAD,CAAC,AA5BD,CAAmC,KAAK,CAAC,SAAS,GA4BjD;AA5BY,sCAAa"}