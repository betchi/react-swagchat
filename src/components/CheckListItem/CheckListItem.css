.sc-check-list-item-root {
  margin-left: 0px;
  position: relative;
  text-decoration: none;
  color: #333;
  height: 60px;
  line-height: 60px;
  display: flex;
  justify-content: space-around;
}
.sc-check-list-item-wrap {
  min-width: 320px;
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  background-color: #fefefe;
  user-select: none;
}
.sc-check-list-item-avatar {
  position: absolute;
}
.sc-check-list-item-subject {
  overflow: hidden;
  text-overflow: ellipsis;
}
.sc-check-list-item-flex1 {
  width: 48px;
  margin: 10px;
}
.sc-check-list-item-flex2 {
  width: 100%;
  margin: 10px;
}
.sc-check-list-item-flex3 {
  width: 50px;
  margin: 10px;
  display: flex;
  justify-content: center;
}
