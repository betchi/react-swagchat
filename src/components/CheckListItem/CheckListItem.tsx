import * as React from 'react';
import { IUser } from 'swagchat-sdk';
import { IOnClickProps } from 'swagchat-sdk';
import { Avatar, Button, RadioButtonChecked, RadioButtonUnChecked } from '../../components';

export interface ICheckListItemProps extends IOnClickProps {
  name: string;
  pictureUrl: string;
  width?: number;
  height?: number;
  isChecked: boolean;
}

export class CheckListItem extends React.Component<ICheckListItemProps, {}> {
  onClick(user: IUser) {
    if (this.props.onClick) {
      this.props.onClick(user);
    }
  }

  render(): JSX.Element  {
    return (
      <div className="check-list-item-wrap" onClick={this.props.onClick}>
        <div className="check-list-item-flex1" style={{height: this.props.width + 'px', lineHeight: this.props.width + 'px'}}>
          <Avatar src={this.props.pictureUrl} width={this.props.width} height={this.props.height} />
        </div>
        <div className="check-list-item-flex2" style={{height: this.props.width + 'px', lineHeight: this.props.width + 'px'}}>
          <div className="check-list-item-subject">{this.props.name}</div>
        </div>
        <div className="check-list-item-flex3" style={{height: this.props.width + 'px', lineHeight: this.props.width + 'px'}}>
          {(() => {
            if (this.props.isChecked) {
              return <Button icon={<RadioButtonChecked />} />;
            } else {
              return <Button icon={<RadioButtonUnChecked style={{fill: 'rgba(153, 153, 153, 0.2)'}} />} />;
            }
          })()}
        </div>
      </div>
    );
  }
}
