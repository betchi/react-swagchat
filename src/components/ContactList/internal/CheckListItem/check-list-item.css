@value colors: "../../../../styles/theme/default/color.css";
@value color-primary, color-grey-alpha from colors;

.root {
  min-width: 320px;
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  background-color: #fefefe;
  user-select: none;
}
.avatar {
  position: absolute;
}
.subject {
  overflow: hidden;
  text-overflow: ellipsis;
}
.flex1 {
  margin: 10px;
}
.flex2 {
  width: 100%;
  margin: 10px;
}
.flex3 {
  width: 50px;
  margin: 10px;
  display: flex;
  justify-content: center;
}

.flex3 .checked {
  color: color-primary;
}
.flex3 .unchecked {
  color: color-grey-alpha;
}