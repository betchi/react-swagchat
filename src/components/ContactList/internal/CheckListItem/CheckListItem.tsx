import * as React from 'react';
import { IOnClickProps } from '../../../';
import * as styles from './check-list-item.css';
const classNames = require('classnames');

export interface ICheckListItemProps extends IOnClickProps {
  text: string;
  icon?: React.ReactNode;
  isChecked?: boolean;
  checkedIcon?: React.ReactNode;
  unCheckedIcon?: React.ReactNode;
  width?: string;
  height?: string;
  fontSize?: string;
  fontColor?: string;
  className?: string;
  style?: Object;
}

export interface ICheckState {
  isChecked: boolean;
}

export class CheckListItem extends React.Component<ICheckListItemProps, ICheckState> {
  public static defaultProps: Partial<ICheckListItemProps> = {
    isChecked: false,
    className: '',
    style: {},
    onClick: () => {},
  };

  constructor(props: ICheckListItemProps) {
    super(props);

    if (props.isChecked) {
      this.state = {isChecked: props.isChecked};
    } else {
      this.state = {isChecked: false};
    }
  }

  onClick() {
    this.setState({isChecked: !this.state.isChecked});
    this.props.onClick ? this.props.onClick() : null;
  }

  render(): JSX.Element  {
    const { text, icon, width, height, fontSize, fontColor } = this.props;

    let rootStyle: {
      width?: string;
      height?: string;
    } = {};
    width ? rootStyle.width = width : null;
    height ? rootStyle.height = height : null;

    let descriptionStyle: {
      color?: string,
      fontSize?: string,
    } = {};
    fontSize ? descriptionStyle.fontSize = fontSize : null;
    fontColor ? descriptionStyle.color = fontColor : null;

    return (
      <div className={styles.root} onClick={this.onClick.bind(this)} style={rootStyle}>
        {icon ? <div className={styles.flex1}>{icon}</div> : null}
        <div className={styles.flex2}>
          <div className={styles.subject} style={descriptionStyle}>{text}</div>
        </div>
        <div className={styles.flex3}>
          {this.state.isChecked ? (
            this.props.checkedIcon ? this.props.checkedIcon : <i className={classNames('material-icons', styles.checked)}>radio_button_checked</i>
          ) : (
            this.props.unCheckedIcon ? this.props.unCheckedIcon : <i className={classNames('material-icons', styles.unchecked)}>radio_button_unchecked</i>
          )}
        </div>
      </div>
    );
  }
}
