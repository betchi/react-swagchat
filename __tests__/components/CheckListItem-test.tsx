declare var jest, describe, it, expect, require;

import * as React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import { CheckListItem } from '../../src/components/ContactList/internal/CheckListItem/CheckListItem';

const testComponent1 = shallow(
  <CheckListItem
    text="Button"
  />
);
it('CheckListItem renders correctly', () => {
  const j = toJson(testComponent1);
  expect(j).toMatchSnapshot();
});

const testComponent2 = shallow(
  <CheckListItem
    text="Button"
    icon={<div />}
    isChecked={true}
    checkedIcon={<div />}
    unCheckedIcon={<div />}
    className="test-class"
    style={{width: '100px'}}
    onClick={() => console.log('click')}
  />
);
it('CheckListItem renders correctly', () => {
  const j = toJson(testComponent2);
  expect(j).toMatchSnapshot();
});
