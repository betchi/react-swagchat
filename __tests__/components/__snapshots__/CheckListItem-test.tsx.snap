// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckListItem renders correctly 1`] = `
<div
  className=""
  onClick={[Function]}
  style={Object {}}
>
  <div>
    <div>
      Button
    </div>
  </div>
  <div>
    <i
      className="material-icons"
    >
      radio_button_unchecked
    </i>
  </div>
</div>
`;

exports[`CheckListItem renders correctly 2`] = `
<div
  className="test-class"
  onClick={[Function]}
  style={
    Object {
      "width": "100px",
    }
  }
>
  <div>
    <div />
  </div>
  <div>
    <div>
      Button
    </div>
  </div>
  <div>
    <div />
  </div>
</div>
`;
